### ISSUE TITLE
MLCore accumulates duplicate config-error violations on every setCode()

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3900]
- Markuplint version: `dev` (v5 RC)
- Parser lang: any
- Node.js version: any
- OS: any

## Describe the bug

`MLCore.#configErrors` grows by one copy of each rule-mapping error every time the document is re-created, and `verify()` reports the whole accumulated array as `config-error` violations. Any caller that reuses an `MLCore`/`MLEngine` instance across code updates — most notably the VS Code extension, which calls `engine.setCode(code)` on every document change — sees the same `config-error` warning duplicated once more per edit.

## Mechanism

1. The `Document` constructor re-generates `ruleset.mappingErrors` on every construction when the ruleset contains a misused namespace wildcard (`packages/@markuplint/ml-core/src/ml-dom/node/document.ts`, nodeRules and childNodeRules branches):

   ```ts
   ruleset.mappingErrors.push(
       new Error(`Namespace wildcard "${ruleName}" only accepts false; use a specific rule name for options`),
   );
   ```

2. `MLCore.#createDocument()` drains them into the instance-lifetime `#configErrors` array:

   ```ts
   this.#configErrors.push(...this.#ruleset.mappingErrors);
   this.#ruleset.mappingErrors.length = 0;
   ```

3. `MLCore.setCode()` calls `#parse()` + `#createDocument()`, so every `setCode()` appends one more copy.

4. `verify()` converts the entire `#configErrors` array into `config-error` violations on every run.

The VS Code server calls `engine.setCode(code)` on every document change (`vscode/src/server/v4.ts` etc.), so the duplicate count equals the number of edits.

## Steps To Reproduce

1. Use a config that produces a mapping error, e.g.:

   ```json
   {
       "nodeRules": [{ "selector": "div", "rules": { "a11y/*": true } }]
   }
   ```

2. Reuse one engine across code updates:

   ```ts
   const engine = new MLEngine(file, options);
   await engine.exec(); // 1 config-error
   await engine.setCode('<div></div>');
   await engine.exec(); // 2 identical config-errors
   await engine.setCode('<div>a</div>');
   await engine.exec(); // 3 identical config-errors
   ```

   Or simply edit a file in VS Code with the markuplint extension and watch the Problems panel.

## The current behavior

The `config-error` violation `Namespace wildcard "a11y/*" only accepts false; ...` is reported N times after N `setCode()` calls.

## The expected behavior

Each distinct mapping error is reported exactly once per verification, regardless of how many times the document has been re-created.

## Suggested direction

`#configErrors` should be derived from the CURRENT document rather than accumulated for the instance lifetime — e.g. reset the mapping-error portion of `#configErrors` at the start of `#createDocument()` before draining, or deduplicate by message.

## Additional context

Found during the code review for the #3890/#3891 fix. That PR truncates `#configErrors` back to its pre-fix state after the multi-pass fix loop, which removes the *within-verify* amplification only — the `setCode()` accumulation path described here is pre-existing and remains.
